pandas/tests/series/test_repr.py

```diff
@@ -202,6 +203,35 @@ def test_latex_repr(self):
 
 class TestCategoricalRepr(object):
 
+    @pytest.mark.skipif(compat.PY3, reason="Decoding failure only in PY2")
+    def test_categorical_repr_unicode(self):
+        # GH#21002 if len(index) > 60, sys.getdefaultencoding()=='ascii',
+        # and we are working in PY2, then rendering a Categorical could raise
+        # UnicodeDecodeError by trying to decode when it shouldn't
+        from pandas.core.base import StringMixin
+
+        class County(StringMixin):
+            name = u'San Sebastián'
+            state = u'PR'
+
+            def __unicode__(self):
+                return self.name + u', ' + self.state
+
+        cat = pd.Categorical([County() for n in range(61)])
+        idx = pd.Index(cat)
+        ser = idx.to_series()
+
+        # set sys.defaultencoding to ascii, then change it back after the test
+        enc = sys.getdefaultencoding()
+        reload(sys)  # noqa:F821
+        sys.setdefaultencoding('ascii')
```

Comment:
there is a context manager this i think

Reply:
There is a context manager for `locale` in pd.util.testing.  Can that be used here or do you have something else in mind?  (I agree it would be prettier)

Reply:
yes pls use that

Reply:
Looks like `tm.set_locale` doesn't change `sys.getdefaultencoding()`.  I could make a new contextmanager specifically for this (which I guess would be a no-op in py3?)